frontend/src/pages/create_campaign/index.tsx

```diff
@@ -80,14 +80,39 @@ const CreateCampaign = () => {
   const onTabChange = (newTab: number) => {
     // only allow user to access review tab if all inputs are non-empty
     if (newTab === reviewTabIdx) {
-      if (
-        campaignName === "" ||
-        description === "" ||
-        cover === null ||
-        questions.length === 0 ||
-        roles.length === 0
-      ) {
-        console.error("All fields must be filled before reviewing!");
+      if (campaignName === "") {
+        pushMessage({
+          message: "Campaign name is required!",
+          type: "error",
+        });
+        return;
+      }
+      if (description === "") {
+        pushMessage({
+          message: "Campaign description is required!",
+          type: "error",
+        });
+        return;
+      }
+      if (cover === null) {
+        pushMessage({
+          message: "Campaign cover is required!",
+          type: "error",
+        });
+        return;
+      }
+      if (questions.length === 0) {
+        pushMessage({
+          message: "At least one question is required!",
+          type: "error",
+        });
+        return;
+      }
```

Comment:
can we also make sure that every role has at least one question? this is smth that trips everyone up a lot. should use `Question.roles` for this: https://github.com/csesoc/chaos/blob/50e40e1e5a1f7376a324ca216c33320d894af3b0/frontend/src/pages/create_campaign/types.ts#L9-L12

Reply:
we should also probably check that the role it's associated with exists (we'll need to fix the role deletion in this page for this later)